### PR TITLE
fix metadata parsing `read_opus(dsn, data_only = TRUE)`

### LINKED ISSUE #104: Error in strsplit(sample_name_long, split = ";") : non-character argument
Hi,

I'm having a strange issue with reading OPUS files and requesting only the data. This used to work, about a year ago:

```r
read_opus('e:/mir/MIR-training/HI-test/spectra/103283XS01.0', data_only = TRUE)
```

The error message looks like this:
```
Error in strsplit(sample_name_long, split = ";") : non-character argument
In addition: Warning messages:
1: In strptime(time_hour, format = "%Y/%m/%d %H:%M:%S", tz = etc_tz) :
  unknown timezone 'Etc/NA'
2: In as.POSIXct.POSIXlt(strptime(time_hour, format = "%Y/%m/%d %H:%M:%S",  :
  unknown timezone 'Etc/NA'
3: In as.POSIXlt.POSIXct(x) : unknown timezone 'Etc/NA'
```

Reading the file works as expected with `data_only = FALSE`.

### PR DESCRIPTION
**Pull request by philipp-baumann**
fixes #104 

- concerns sample name and correctly getting timezone info when `data_only = TRUE`. `data_only` is not strictly true, since we also get non-data blocks (sample and history). Still, I think it is a good default to do so and uniquely identify FT-IR measurements/replicates (sample name that was originally entered/correct time, since file operations of the OS change these attributes).

- added new tests to check for both warnings and errors, both for `read_opus(dsn, data_only = FALSE)` and `read_opus(dsn, data_only = TRUE)`, in all example files.

- incorporate suggestion by @mtalluto to get rid of the warning in `strptime` due to extra tab.

To summarize, `data_only = TRUE` condition missed "sample" and "history" blocks when trying to extract metadata. In the result data list, these are removed. Instead, the relevant info extracted is in `basic_metadata` list element.

### CODE REVIEW

**Comment by philipp-baumann** [comment]
we can ignore the code check, that is solved by automatic fixes from pre-commit CI. Also, the extra tests for warnings and `data_only = TRUE` pass, locally and in CI.

Local confirmation:

```r
> devtools::test()
ℹ Testing opusreader2
✔ | F W  S  OK | Context
✔ |          6 | create_dataset                                                                                       
✔ |          4 | parse_header                                                                                         
✔ |          4 | read_opus                                                                                            

══ Results ═══════════════════════════════════════════════════════════════════════════════════════════════════════════
[ FAIL 0 | WARN 0 | SKIP 0 | PASS 14 ]
```

**Comment by philipp-baumann** [comment]
okay, squashed in the pre-commit lite commit, via `git rebase -i HEAD~2` and force-pushed, so you won't see it here any more and the CI is relaunched and gets green status.

**Comment by philipp-baumann** [comment]
once this is merged, we can release patch opusreader2 v0.6.3
